software/FossaSat2/src/ADCS/ADS/kalman_filtering.cpp

```diff
@@ -74,8 +88,8 @@ void ADS_Inverse_Matrix(ADCS_CALC_TYPE matrix[ADCS_STATE_DIM][ADCS_STATE_DIM]) {
 
 /*************** Main function ******************/
 void ADS_Kalman_Filter(const ADCS_CALC_TYPE Q, const ADCS_CALC_TYPE R, const ADCS_CALC_TYPE delta_t,
-                       const ADCS_CALC_TYPE x0[ADCS_STATE_DIM], const ADCS_CALC_TYPE y0[ADCS_STATE_DIM], const ADCS_CALC_TYPE u0[ADCS_STATE_DIM],
-                       ADCS_CALC_TYPE P[ADCS_STATE_DIM][ADCS_STATE_DIM], ADCS_CALC_TYPE filtered_y[ADCS_STATE_DIM]) {
+                       const ADCS_CALC_TYPE x0[], const ADCS_CALC_TYPE y0[], const ADCS_CALC_TYPE u0[],
+                       ADCS_CALC_TYPE P[][ADCS_STATE_DIM], ADCS_CALC_TYPE filtered_y[]) {
```

Comment:
Please keep the fully specified array dimensions, they let our static code checking tool catch out-of-bounds access.